packages/mathjax-extension/src/index.ts

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+/**
+ * @packageDocumentation
+ * @module mathjax-extension
+ */
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+
+import { ILatexTypesetter } from '@jupyterlab/rendermime';
+
+import type { MathDocument } from 'mathjax-full/js/core/MathDocument';
+
+enum CommandIDs {
+  copy = 'mathjax:clipboard',
+  scale = 'mathjax:scale'
+}
+
+/**
+ * The MathJax Typesetter.
+ */
+export class MathJaxTypesetter implements ILatexTypesetter {
+  constructor(app: JupyterFrontEnd) {
+    app.commands.addCommand(CommandIDs.copy, {
+      execute: (args: any) => {
+        const md = this._mathDocument;
+        const oJax: any = md.outputJax;
+        navigator.clipboard.writeText(oJax.math.math);
+      },
+      label: 'MathJax Copy Latex'
+    });
+
+    app.commands.addCommand(CommandIDs.scale, {
+      execute: (args: any) => {
+        const scale = args['scale'] || 1.0;
+        const md = this._mathDocument;
+        md.outputJax.options.scale = scale;
+        md.rerender();
+      },
+      label: args =>
+        'Mathjax Scale ' + (args['scale'] ? `x${args['scale']}` : 'Reset')
+    });
+  }
+
+  protected async _ensureInitialized() {
+    if (this._initialized) {
+      return;
+    }
+
+    await import('mathjax-full/js/input/tex/require/RequireConfiguration');
```

Comment:
As mentioned by @krassowski, I do think to that we should not import `mathjax-full` but the lighter `mathjax`. I saw that it brings question. But we should spend time trying to use the lighter version.

Reply:
1. This should have no impact on initial bundle size (and by extension, initial load times), because `mathjax-full` is loaded dynamically.

2. There is a potential, unknown security risk with removing the line

```
mathjax.handlers.register(SafeHandler(new HTMLHandler(browserAdaptor())));
```

which requires `mathjax-full`, as these are internal undocumented library classes. Replacing `mathjax-full` without first consulting somebody more knowledgable on this (e.g. the extension contributors or MathJax contributors) is dangerously hasty and imposes risk to users.

3. Part of the large difference in package sizes in `mathjax` and `mathjax-full` is just the presence of comprehensive declaration files that are entirely missing in `mathjax`. We must be careful to distinguish **package size** and **impact on bundle size**. Declaration files increase package size but have no impact on bundle size. Build optimizations such as dependency deduping complicate this further, as even library code included in a package may have no impact on bundle size.

4. If we discover a way to safely replace `mathjax-full` with `mathjax`, this would not be a breaking change, and hence can be addressed in a follow-up PR.